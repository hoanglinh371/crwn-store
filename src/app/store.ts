import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
