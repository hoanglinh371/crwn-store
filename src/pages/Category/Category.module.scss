.container {
  display: flex;
  flex-direction: column;
}

.products {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  column-gap: 20px;
  row-gap: 50px;
}

.title {
  font-family: cursive;
  font-size: 28px;
  margin-bottom: 25px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
}

@media screen and (min-width: 0) and (max-width: 600px) {
  .container {
    .title {
      font-size: 20px;
      margin-bottom: 12px;
    }

    .products {
      grid-template-columns: 1fr 1fr;
      row-gap: 32px;
      column-gap: 10px;
    }
  }
}
