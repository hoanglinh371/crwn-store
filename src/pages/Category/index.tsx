import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from 'src/app/hooks';
import classNames from 'classnames/bind';

import styles from './Category.module.scss';

import ProductCard from 'src/components/ProductCard';
import { selectProducts } from 'src/features/product/redux/product.selectors';
import { getProductsByCategoryThunk } from 'src/features/product/redux/product.thunks';

const cx = classNames.bind(styles);

const Category: React.FC = () => {
  const { category } = useParams();
  const dispatch = useAppDispatch();
  const products = useAppSelector(selectProducts);

  useEffect(() => {
    dispatch(getProductsByCategoryThunk(category as string));
  }, [dispatch, category]);

  return (
    <div className={cx('container')}>
      <h2 className={cx('title')}>{category}</h2>
      <div className={cx('products')}>
        {products &&
          products.map((product) => (
            <ProductCard key={product._id} product={product} />
          ))}
      </div>
    </div>
  );
};

export default Category;
