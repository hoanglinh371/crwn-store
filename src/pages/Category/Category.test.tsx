import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from 'src/app/store';
import Category from '.';

describe('Category component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Category />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
