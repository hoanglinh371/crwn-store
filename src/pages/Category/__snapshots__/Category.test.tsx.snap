// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="title"
    />
    <div
      class="products"
    />
  </div>
</div>
`;
